### PR TITLE
US-5.2: Show directions for selected outdoor point of interest

### LINKED ISSUE #24: US-5.2: Show directions for selected outdoor point of interest
As a user, I want to have directions to the selected outdoor point of interest

**Acceptance criteria:**

- [x] User can select a POI from list or map.
- [x] Directions are generated correctly.
- [x] Route is shown on the map.

### LINKED ISSUE #291: AT-5.2: Show directions for selected outdoor point of interest
**Acceptance criteria:**

- [ ] User can select a POI from list or map.
- [ ] Directions are generated correctly.
- [ ] Route is shown on the map.

### PR DESCRIPTION
**Pull request by khalillabban**
- Implemented outdoor POI directions from both list and map pin selection.
- Implemented a selected POI panel with Get directions and Clear actions.
- Implemented POI destination routing with current-location start + fallback.
- Implemented POI route recalculation when the selected destination changes.
- Implemented user-facing error handling for POI routing failures.
- Implemented tests for the new POI routing flow and edge cases.
image

Closes #24
Closes #291

### CODE REVIEW

File: app/CampusMapScreen.tsx [hunk 1]
```diff
@@ -513,6 +520,28 @@ export default function CampusMapScreen() {
...
+  const handleSelectOutdoorPOI = useCallback((poi: PlacePOI) => {
+    setSelectedOutdoorPOI(poi);
+    setPOIRouteError(null);
+    setFocusPOIId(poi.placeId);
+    setFocusPOITrigger((c) => c + 1);
+    setShowPOIList(false);
+
+    setActiveOutdoorPOIRoute((previousRoutePOI) => {
+      if (!previousRoutePOI) return previousRoutePOI;
+      if (previousRoutePOI.placeId === poi.placeId) return previousRoutePOI;
```
Comment (NicG27) [inline app/CampusMapScreen.tsx]: Change to this please:
```
if (!previousRoutePOI || previousRoutePOI.placeId === poi.placeId) return previousRoutePOI;
```

File: app/CampusMapScreen.tsx
Comment (NicG27) [inline app/CampusMapScreen.tsx]: Please remove this comment

Comment (NicG27) [inline app/CampusMapScreen.tsx]: Please remove this comment

File: app/CampusMapScreen.tsx [hunk 2]
```diff
@@ -759,9 +802,38 @@ export default function CampusMapScreen() {
   }, [mergedSteps, routeSteps, canContinueIndoors, continueIndoorsBuildingCode, destinationRoomQueryText, openIndoorMap]);
 
   // ── Render ─────────────────────────────────────────────────────────────────
```
[the remark this answers is not in the document]
Comment (khalillabban) [reply app/CampusMapScreen.tsx]: Done

File: app/CampusMapScreen.tsx
Comment (khalillabban) [reply app/CampusMapScreen.tsx]: Done
